### ISSUE TITLE
Outdated codes on for building ngx-extended-pdf-viewer

### ISSUE THREAD

**Issue by chan-dev** [stephanrauh/ngx-extended-pdf-viewer#905]
I've followed the steps [here](https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/main/projects/ngx-extended-pdf-viewer/how-to-build.md) for building custom pdfs.js.

There are some outdated codes when cloning the repo.
For example:
1. `@angular/material` is not installed
2. This may seem obvious but what should be the value of `/path/` here? 
Running the app shows these errors
![pdfjs-error](https://user-images.githubusercontent.com/46891115/132200225-acc0c8d3-4ecf-42f5-af4d-ee1e369af101.png)

```
start": "ng serve --deployUrl=/path/ --host 10.0.1.3 --disable-host-check",
```

Apart from those, building the pdf.js after `gulp serve`, it shows an error
![pdfjs-gulp](https://user-images.githubusercontent.com/46891115/132200386-3922ce6d-26c4-4ec8-8ff0-54db610eaaf9.png)

**Comment by stephanrauh** [comment]
I've stopped using `gulp server` and the default demo of the project. So you're right, I have to update the documentation.

BTW, "path" is literally "path". I put the default demo into a context path to make sure the library works both in the root context and in any other context. Such as "localhost:4200/path".

My current workflow is:
- cd mypdf.js
- git checkout bleeding-edge (or git checkout 2.10)
- npm i
- gulp generic (just to make sure everything OK)

After that, I start editing the source files of pdf.js:

- cd ../ngx-extended-pdf-viewer
- ./updateMozillasPdfViewer.sh (the Windows counterpart also works, but it may be slightly out-of-date!)
- npm run showcase
- open a second terminal and start the showcase with `ng s -o`

**Comment by stephanrauh** [comment]
BTW, canvas-size is a dependency in the package.json. But I deleted the test folder of pdf.js including the test server, so `gulp server` doesn't work, anyways.

**Comment by stephanrauh** [comment]
Please have a look at the updated instructions. It's always hard to write instructions - which are basically an algorithm in plain English, without compiler and debugger - without mistakes, so I appreciate your feedback.

Best regards,
Stephan

**Comment by chan-dev** [comment]
Many thanks for being so helpful and responsive. I'll keep you updated once i'm able to finish the custom build or if there are any issues on my end.

**Comment by chan-dev** [comment]
Minor comments with the updated instructions:

I've followed the instructions and there were some few errors after step 5.
The way i was able to run the showcase was with the following steps:

### Use the showcase to test your changes of ngx-extended-pdf-viewer

Make sure you clone each respective repo and run `npm install` upon running `git clone`

NOTE: after every `git clone` , `cd` to that repo and run `npm install`

1. `mkdir -p containing-folder`
2. `cd containing-folder`
3. `git clone https://github.com/stephanrauh/ngx-extended-pdf-viewer`
4. Create embedded-pdf folder and cd to it: `mkdir -p embedded-pdf && cd embedded-pdf`
5. `git clone https://github.com/stephanrauh/extended-pdf-viewer-showcase`
6. `git clone https://github.com/stephanrauh/ngx-extended-pdf-viewer`
7. After running all above commands, the directory tree should look like this:

    .
    └── containing-folder/
        ├── embedded-pdf/
        │   ├── extended-pdf-viewer-showcase
        │   └── ngx-extended-pdf-viewer
        └── ngx-extended-pdf-viewer

8. Then cd back to `ngx-extended-pdf-viewer` and make sure you're in `containing-folder/embedded-pdf/ngx-extended-pdf-viewer`
9. Run `npm run showcase`
10. Open a new terminal, cd to `containing-folder/embedded-pdf/extended-pdf-viewer-showcase`
11. Run `ng serve -o`

Kindly check if there are any steps that needs to be done, but in my case i was able to make it work.

**Comment by chan-dev** [comment]
> My current workflow is:
> 
> * cd mypdf.js
> * git checkout bleeding-edge (or git checkout 2.10)
> * npm i
> * gulp generic (just to make sure everything OK)
> 
> After that, I start editing the source files of pdf.js:
> 
> * cd ../ngx-extended-pdf-viewer
> * ./updateMozillasPdfViewer.sh (the Windows counterpart also works, but it may be slightly out-of-date!)
> * npm run showcase
> * open a second terminal and start the showcase with `ng s -o`

Regarding the workflow, I'm editing the pdf.js repo and commenting out some lines but they don't get reflected on the showcase.

**Comment by chan-dev** [comment]
I've found the reason why the changes are not reflecting.
Instead of checkout out `bleeding-edge` branch, i've checked out `2.10` branch.

As a result, the `./updateMozillasPdfViewer.sh` **always** default to `assets` folder because of certain condition in shells script.

Before it was, as you can see there was no checking for branch `2.10`
```
if [ "$BRANCH" == "bleeding-edge" ]; then
  FOLDER="bleeding-edge"
fi
```

Second issue was when i'm running it, there's an error `[[` was not found. 

To make the script work on my end, i've updated it with the following changes:
Changes that i did were:
1. Use `bash` instead of `sh`, i don't know if that really makes a difference
2. Add condition for checking `2.10` branch
3. Add `echo` to display that we're on the correct branch
```
#!/bin/bash
cd ../mypdf.js

FOLDER="assets"
BRANCH=$(git rev-parse --abbrev-ref HEAD)
if [ "$BRANCH" == "bleeding-edge" ] || [ "$BRANCH" == "2.10" ]; then
  FOLDER="bleeding-edge"
fi

echo Folder is "$FOLDER"
```

**Comment by stephanrauh** [comment]
If everything works as intended, the files are copied
- to the assets folder if you've checked out the 2.10 branch,
- to the "bleeding edge" folder if you've checkout out the (guess what!) bleeding_edge branch.

2.10 is what I call the stable branch, although the pdf.js team still considers it a preview.

**Comment by stephanrauh** [comment]
(reopened because I want to add the improvements and suggestions above)

**Comment by stephanrauh** [comment]
It took a while... but finally, I've found the time to update the build instructions.

Enjoy!
Stephan
